ui/forms/MainWindow.qml

```diff
@@ -127,34 +128,48 @@ ApplicationWindow {
     text: `Zoom: ${(systemPreview.zoom * 100).toFixed(0)}%`
   }
 
-  Button {
+  RowLayout {
     anchors.bottom: zoomLabel.top
     anchors.left: zoomLabel.left
     anchors.bottomMargin: Theme.padding
-    text: "Debug Source"
-    outlined: true
 
-    onClicked: {
-      astInspectorLoader.active = true;
-    }
+    spacing: Theme.units(4)
 
-    Loader {
-      id: astInspectorLoader
-      active: false
-      sourceComponent: AstInspector {
-        anchors.centerIn: Overlay.overlay
-        modelSource: documentModel.debugSource
-        astSource: documentModel.astSource
+    Button {  
+      text: "Debug Source"
+      // outlined: true
 
-        onClosed: {
-          astInspectorLoader.active = false;
+      onClicked: {
+        astInspectorLoader.active = true;
+      }
+
+      Loader {
+        id: astInspectorLoader
+        active: false
+        sourceComponent: AstInspector {
+          anchors.centerIn: Overlay.overlay
+          modelSource: documentModel.debugSource
+          astSource: documentModel.astSource
+
+          onClosed: {
+            astInspectorLoader.active = false;
+          }
         }
       }
     }
+
+    Button {
+      text: systemPreview.running ? "Pause" : "Run"
+      icon.source: systemPreview.running ? "qrc:/icons/pause.svg" : "qrc:/icons/play.svg"
+
+      onClicked: {
+        systemPreview.running = !systemPreview.running
+      }
+    }
   }
 
   Fab {
-    anchors.right: parent.right
+    anchors.right: documentEditor.left
```

Comment:
Do I understand it right that this effectively makes the fab move along with the sidebar?

Reply:
Yes, I thought it looks a bit nicer when sidebar is expanded, but I can revert it if you want.

Reply:
It's fine to keep it, thanks!